Sources/Regex/Compiler.swift

```diff
@@ -0,0 +1,132 @@
+import MatchingEngine
+
+public struct RegexProgram {
+  typealias Program = MatchingEngine.Program<String>
+  var program: Program
+}
+
+public class Compiler {
+  public let ast: AST
+  public let options: REOptions
+  private var builder = RegexProgram.Program.Builder()
+
+  public init(ast: AST, options: REOptions = []) {
+    self.ast = ast
+    self.options = options
+  }
+
+  public __consuming func emit() -> RegexProgram {
+    emit(ast)
+    builder.buildAccept()
+    return RegexProgram(program: builder.assemble())
+  }
+
+  func emit(_ node: AST) {
+    switch node {
+    case .any:
+      builder.buildConsume(1)
+
+    case .alternation(let components):
+      let done = builder.createAddress()
+      for component in components.dropLast() {
+        let next = builder.createAddress()
+        builder.buildSave(next)
+        emit(component)
+        builder.buildBranch(to: done)
+        builder.label(next)
+      }
+      emit(components.last!)
+      builder.label(done)
+
```

Comment:
For the more complicated cases, could you provide a comment showing what the generate code looks like?

Reply:
Added.